sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -93,18 +107,26 @@ def _renewable(self, renewable):
             return False
         return True
 
-    def _auto_lock_renew(self, receiver, renewable, starttime, timeout, on_lock_renew_failure=None):
+    def _auto_lock_renew(self,
+                         receiver,
+                         renewable,
+                         starttime,
+                         max_lock_renewal_duration,
+                         on_lock_renew_failure=None,
+                         renew_period_override=None):
         # pylint: disable=protected-access
-        _log.debug("Running lock auto-renew thread for %r seconds", timeout)
+        _log.debug("Running lock auto-renew thread for %r seconds", max_lock_renewal_duration)
         error = None
         clean_shutdown = False  # Only trigger the on_lock_renew_failure if halting was not expected (shutdown, etc)
         try:
             while self._renewable(renewable):
-                if (utc_now() - starttime) >= datetime.timedelta(seconds=timeout):
-                    _log.debug("Reached auto lock renew timeout - letting lock expire.")
-                    raise AutoLockRenewTimeout("Auto-renew period ({} seconds) elapsed.".format(timeout))
-                if (renewable.locked_until_utc - utc_now()) <= datetime.timedelta(seconds=self._renew_period):
-                    _log.debug("%r seconds or less until lock expires - auto renewing.", self._renew_period)
+                if (utc_now() - starttime) >= datetime.timedelta(seconds=max_lock_renewal_duration):
+                    _log.debug("Reached max auto lock renew duration - letting lock expire.")
+                    raise AutoLockRenewTimeout("Auto-renew period ({} seconds) elapsed.".format(
+                        max_lock_renewal_duration))
+                renew_period = renew_period_override or self._renew_period
```

Comment:
minor optimization: this line could be moved out of the while loop

Reply:
more than optimization, the code reads far clearer when it's moved as you describe.  I had mindlessly followed "put instantiation near utilization" but your point is 100%++